### DISCUSSION TITLE
친구 제한 수의 범위 결정

### DISCUSSION THREAD

**Discussion by jis-kim** [GhostPangPang/GhostPong#56] [General]
현재 저희 서비스는 42명의 친구 제한을 두고 있는데요, 
친구 제한을 할 것인지 친구 신청 제한을 할 것인지가 불분명합니다. 일단 현재는 친구 수 + 친구 제한 수 합쳐서 42명 제한 로직입니다!
## 1. 그대로 갈 경우
친구 신청 할 때의 쿼리가 간단합니다. (내가 받은 신청이 수락 여부 상관없이 42명을 넘었으면 에러)
친구 수락 할 때가 애매해집니다. 친구 수락 시 신청을 보낸 사람의 친구 수가 42명인가? 를 체크 해야 하는데 친구 수를 체크하는 것은 말이 되지만 친구 신청받은 수를 체크하는 것은 말이 안되죠.. 예를 들어 소현 -> 한나 신청을 보내놓고 소현이 받은 친구 신청 수 + 현재 친구 수가 42명이 넘어버린다면 한나가 수락하지 못하는 상황이 되어 버립니다. 😅 이상하다 이상해..

## 2. 나눠서 갈 경우
친구 신청을 보낼 때 보내는 상대방의 1. 친구가 다 찼는가 2. 친구 신청이 다 찼는가 를 체크하고 보내야 합니다. 받을 때는 상대방의 친구가 다 찼는가만 체크하면 됩니다. 이건 지금 문제점 생각이 잘 안나는군요...

**Comment by anso33** [comment]
1번 이상하긴 하네요 😅
저는 2번이 맞말이라고 생각합니다. -> 친구 제한수 친구 신청 제한 수 정해야 함

**Reply by jis-kim** [reply]
의견 고맙습니다! 2번으로 결정나면 일단 42/42 설정해두고 구현하겠습니닷

**Comment by Kimhan-nah** [comment]
1번으로 유지하려고 했던 건 count하는 로직이 1번 더 필요해서 였는데 1번으로 유지해도 문제점이 발생한 거면 2번으로 가는 게 정석이기도 하고 더 쉬워지지 않을까여?? 2번이 더 깔꼼할 거 같아용

**Reply by jis-kim** [reply]
그러게용.. 2번은 근데 현재로써 많이 생각해보지 않아서..ㅋㅋㅋ 문제점 생각나면 다시 업데이트 하겠습니다.

**Comment by aLVINlEE9** [comment]
저도 2번 추천입니다

**Reply by jis-kim** [reply]
3인의 동의로... 2번으로 가보겠습니다 고맙습니다!
